### PR TITLE
upgrade to flow 0.53.0 & fix lint errors

### PR DESCRIPTION
**Pull request by Palisand**
https://github.com/react-native-community/react-native-tab-view/issues/359

There are no more flow or linting errors.

@satya164 Please consider merging this request.

### CODE REVIEW

**Comment by Palisand** [comment]
0.49.2 has taken care of these 2 issues, will update soon.

**Comment by Ashoat** [comment]
@Palisand - maybe try running `eslint --fix .` in the root directory to fix the Prettier error?

@satya164 - could you take a look at this? Flow 0.53 has breaking changes for typing React components, and these changes are necessary to get React Navigation working with Flow 0.53.

File: src/TabViewPagerAndroid.js [hunk 1]
```diff
@@ -94,13 +94,14 @@ export default class TabViewPagerAndroid<T: Route<*>> extends PureComponent<
 
   _setPage = (index: number) => {
     if (this._viewPager) {
+      // in spite of this check, flow 0.53.0 still reports errors if `this._viewPager &&` is omitted below
```
Comment (Ashoat) [inline src/TabViewPagerAndroid.js]: Flow doesn't allow you to refine properties on `this`. To get around this, assign to a locally scoped variable, and  refine the type of the locally scoped variable.

```js
const viewPager = this._viewPager;
if (viewPager) {
```

File: src/TouchableItem.js [hunk 2]
```diff
@@ -18,19 +19,11 @@ type Props = {
...
-type DefaultProps = {
-  pressColor: string,
-};
-
-export default class TouchableItem extends PureComponent<
-  DefaultProps,
-  Props,
-  void
-> {
+export default class TouchableItem extends PureComponent<Props, void> {
```
Comment (sibelius) [inline src/TouchableItem.js]: <Props> is enough

File: src/TabViewPagerPan.js [hunk 3]
```diff
@@ -72,7 +67,6 @@ const DefaultTransitionSpec = {
 };
 
 export default class TabViewPagerPan<T: Route<*>> extends PureComponent<
-  DefaultProps,
   Props<T>,
   void
```
Comment (sibelius) [inline src/TabViewPagerPan.js]: no need for void here

File: src/TouchableItem.js (hunk 2 above)
Comment (Palisand) [reply src/TouchableItem.js]: What is enough? Based on your other comment, I'm guessing `<Props>` is enough, yes?

**Comment by Palisand** [comment]
I will:
- remove all unnecessary occurrences of `void` `State`s
- add the type refinement fix
- include eslint fixes

File: src/TouchableItem.js (hunk 2 above)
Comment (sibelius) [reply src/TouchableItem.js]: YEp

**Comment by Palisand** [comment]
Still not sure why Travis is reporting the same 49 lint errors. Running `npm run lint` or `yarn run lint` locally results in no errors. I'm going to close and open to trigger another build.

**Comment by satya164** [comment]
I've cleared the cache on CI and triggered a rebuild. Sorry I've not been able to review it. Been a bit busy. Will review soon.

**Comment by Ashoat** [comment]
@palisand - you definitely ran `eslint --fix .` right? The Prettier Lint errors are still showing up. When you run `eslint .`, do the same errors show up as in the CI build?

**Comment by Palisand** [comment]
@Ashoat No errors are reported after running `npm run lint` on my machine. I'm going to reset and make the change to fix [this error reported by the build for the `flow fixes` commit](https://travis-ci.org/react-native-community/react-native-tab-view/builds/286713345?utm_source=github_status&utm_medium=notification) (which I should've done in the first place). Do you know what could be causing the local-travis eslint mismatch?

**Comment by Palisand** [comment]
FINALLY... still getting 48 errors after `eslint .` though.

**Comment by Ashoat** [comment]
Awesome! @satya164, can we get this merged now?

**Comment by Ashoat** [comment]
Ping! @satya164, can we merge now that CI is passing?

**Comment by Ashoat** [comment]
@satya164, this PR is now actively blocking react-navigation from upgrading RN versions. Please take a second to merge it!

**Comment by Palisand** [comment]
@satya164 @Ashoat Thanks guys!

**Comment by satya164** [comment]
I have released a new version. Thanks for all your hard work! ❤️

**Pull request merged**